source/framework/core/src/TRestDataSet.cxx

```diff
@@ -103,6 +103,10 @@
 ///    // Will add to the final tree only the specific observables
 ///    <observables list="g4Ana_totalEdep:hitsAna_energy" />
 ///
+///    // Will apply a cut to the observables
```

Comment:
I think that to re-use `TRestCut` inside `TRestDataSet` is the way to go but it would require some modifications inside `TRestCut` .

Personally, I don't like the way that metadata filters for run parameters are handled inside `TRestDataSet`. I think we should think about a better way to do it.

Reply:
Which kind of modification in `TRestCut`?

Reply:
Please check https://github.com/rest-for-physics/framework/pull/391

Feel free to merge these changes into this branch